code/game/objects/structures/hand_cart.dm

```diff
@@ -0,0 +1,70 @@
+/obj/structure/hand_cart
+	name = "handcart"
+	desc = "A wheeled cart used to make heavy things less difficult to move through the power of lever-arm and the wheel."
+	icon = 'icons/obj/structures/handcart.dmi'
+	icon_state = "cart"
+	anchored = 0
+	density = 1
+	atom_flags = ATOM_FLAG_WHEELED
+	material_alteration = MAT_FLAG_ALTERATION_NAME | MAT_FLAG_ALTERATION_COLOR
+	material = /decl/material/solid/metal/steel
+	w_class = ITEM_SIZE_NO_CONTAINER
+	color = COLOR_ORANGE
+	var/pixel_y_offset = 4
+	var/pixel_x_offset = 4
+	var/atom/movable/carrying
+	var/min_object_size = ITEM_SIZE_NORMAL
+
+/obj/structure/hand_cart/on_update_icon()
+	underlays.Cut()
+	cut_overlays()
+	var/image/W = image(icon, "cart_wheel")
+	underlays += W
+	var/image/I = image(icon, "handcart_layer_north")
+	I.layer = STRUCTURE_LAYER + 0.02
+	I.color = BlendRGB(color, material.color, 0.5)
+	add_overlay(I)
+	if(carrying)
+		var/image/CA = image(carrying.icon, carrying.icon_state)
+		CA.pixel_y = pixel_y_offset
+		CA.pixel_x = pixel_x_offset
+		CA.plane = plane
+		CA.layer = layer + 0.01 //just above STRUCTURE_LAYER
+
+/obj/structure/hand_cart/Destroy()
+	. = ..()
+	if(carrying)
+		carrying.forceMove(get_turf(src))
+
+/obj/structure/hand_cart/grab_attack(var/obj/item/grab/G)
+	if(G.affecting && istype(G.affecting, /obj/))
+		to_chat(G.assailant, SPAN_NOTICE("You start loading \the [G.affecting] onto \the [src]."))
+		if(load_item(G.affecting, G.assailant))
+			qdel(G)
```

Comment:
`return TRUE` possibly.

Reply:
Definitely needs some kind of return to prevent the rest of the grab logic acting on a qdeleted grab.